pkg/kv/kvserver/rafttrace/rafttrace_test.go

```diff
@@ -0,0 +1,298 @@
+
+	ent := raftpb.Entry{Index: 1}
+	rt.MaybeRegister(ctx, ent)
+
+	msg := raftpb.Message{
+		Term:  1,
+		From:  1,
+		To:    2,
+		Type:  raftpb.MsgAppResp,
+		Index: uint64(5),
+	}
+
+	tracedEntries := rt.MaybeTrace(msg)
+	require.Len(t, tracedEntries, 0)
+
+	output := finish().String()
+	require.NoError(t, testutils.MatchInOrder(output, []string{"1->2 MsgAppResp Term:1 Index:5"}...))
+	require.Equal(t, int64(1), rt.numRegisteredStore.Load())
+}
+
+func TestDupeMsgAppResp(t *testing.T) {
+	rt := createTracer(10)
+	ctx, finish := tracing.ContextWithRecordingSpan(context.Background(), rt.tracer, "test")
+
+	ent := raftpb.Entry{Index: 1}
+	rt.MaybeRegister(ctx, ent)
+
+	// Second one should not trace.
+	rt.MaybeTrace(
+		raftpb.Message{
+			Term:  1,
+			From:  1,
+			To:    2,
+			Type:  raftpb.MsgAppResp,
+			Index: uint64(5),
+		})
+	rt.MaybeTrace(
+		raftpb.Message{
+			Term:  1,
+			From:  1,
+			To:    2,
+			Type:  raftpb.MsgAppResp,
+			Index: uint64(6),
+		})
+
+	output := finish().String()
+	require.NoError(t, testutils.MatchInOrder(output, []string{"1->2 MsgAppResp Term:1 Index:5"}...))
+	require.Error(t, testutils.MatchInOrder(output, []string{"1->2 MsgAppResp Term:1 Index:6"}...))
+	require.Equal(t, int64(1), rt.numRegisteredStore.Load())
+}
+
+func TestTraceMsgStorageApplyResp(t *testing.T) {
+	rt := createTracer(10)
+	ctx, finish := tracing.ContextWithRecordingSpan(context.Background(), rt.tracer, "test")
+
+	ent := raftpb.Entry{Index: 1}
+	rt.MaybeRegister(ctx, ent)
+
+	msg := raftpb.Message{
+		From:    1,
+		To:      2,
+		Term:    3,
+		Type:    raftpb.MsgStorageAppendResp,
+		Index:   uint64(5),
+		LogTerm: uint64(4),
+	}
+
+	tracedEntries := rt.MaybeTrace(msg)
+	require.Len(t, tracedEntries, 0)
+
+	output := finish().String()
+	require.NoError(t, testutils.MatchInOrder(output, []string{"1->2 MsgStorageAppendResp Log:4/5"}...))
+	require.Equal(t, int64(1), rt.numRegisteredStore.Load())
+}
+
+func TestNoTraceMsgStorageApplyResp(t *testing.T) {
+	rt := createTracer(10)
+	ctx, finish := tracing.ContextWithRecordingSpan(context.Background(), rt.tracer, "test")
+
+	ent := raftpb.Entry{Index: 10}
+	rt.MaybeRegister(ctx, ent)
+
+	msg := raftpb.Message{
+		From:    1,
+		To:      2,
+		Term:    3,
+		Type:    raftpb.MsgStorageAppendResp,
+		Index:   uint64(5),
+		LogTerm: uint64(4),
+	}
+
+	// Doesn't trace since the index is behind the entry index.
+	rt.MaybeTrace(msg)
+
+	output := finish().String()
+	require.Error(t, testutils.MatchInOrder(output, []string{"MsgStorageAppendResp"}...))
+	require.Equal(t, int64(1), rt.numRegisteredStore.Load())
+}
+
+func TestTraceMsgStorageApply(t *testing.T) {
+	rt := createTracer(10)
+	ctx, finish := tracing.ContextWithRecordingSpan(context.Background(), rt.tracer, "test")
+
+	rt.MaybeRegister(ctx, raftpb.Entry{Index: 1})
+
+	msg := raftpb.Message{
+		From:    1,
+		To:      2,
+		Term:    3,
+		Type:    raftpb.MsgStorageApplyResp,
+		Index:   uint64(5),
+		LogTerm: uint64(4),
```

Comment:
`Term`, `Index`, and `LogTerm` for this type of message are unused. Remove?

https://github.com/cockroachdb/cockroach/blob/b2e1d9a00915b6e835893120ff029592755c48ae/pkg/raft/rawnode.go#L459-L465

Reply:
Done